src/zc/zodbdgc/README.test

```diff
@@ -726,6 +726,219 @@ files directly:
+
+Next, we add a persistent object to the first database:
+
+    >>> conn1 = db.open()
+    >>> conn2 = conn1.get_connection('db2')
+    >>> pers = conn1.root.x = C()
+    >>> conn1.root.x_alias = pers
+    >>> conn1.add(pers)
+
+We can make a weak reference to this object in both databases
+as well as a strong cross-DB reference:
+
+    >>> from persistent.wref import WeakRef
+    >>> conn1.root.wx = WeakRef(pers)
+    >>> conn2.root.wx = WeakRef(pers)
+    >>> transaction.commit()
+    >>> conn2.root.x = pers
+    >>> transaction.commit()
+
+Time passes. :)
+
+    >>> now += 7 * 86400        # 7 days
+
+The number of objects in the databases now:
+
+    >>> len(conn1._storage), len(conn2._storage)
+    (2, 1)
+
+Packing doesn't change it:
+
+    >>> for d in db.databases.values():
+    ...     d.pack()
+    >>> len(conn1._storage), len(conn2._storage)
+    (2, 1)
+    >>> _ = [d.close() for d in db.databases.values()]
+
+We can GC and nothing is collected:
+
+    >>> zc.zodbdgc.gc('config', days=2)
+    db1: roots
+    db1: recent
+    db2: roots
+    db2: recent
+    db1: remove garbage
+    Removed 0 objects from db1
+    db2: remove garbage
+    Removed 0 objects from db2
+
+Now we can delete the object and jump ahead:
+
+    >>> with open('config', 'r') as f:
+    ...     db = ZODB.config.databaseFromFile(f)
+    >>> conn1 = db.open()
+    >>> conn2 = conn1.get_connection('db2')
+    >>> del conn1.root.x
+    >>> transaction.commit()
+    >>> now += 7 * 86400        # 7 days
+    >>> len(conn1._storage), len(conn2._storage)
+    (2, 1)
+    >>> db.pack()
+    >>> len(conn1._storage), len(conn2._storage)
+    (2, 1)
+    >>> _ = [d.close() for d in db.databases.values()]
+
+Still no garbage (the alias holds onto it) is collected:
+
+    >>> now += 7 * 86400        # 7 days
+    >>> zc.zodbdgc.gc('config', days=2)
+    db1: roots
+    db1: recent
+    db2: roots
+    db2: recent
+    db1: remove garbage
+    Removed 0 objects from db1
+    db2: remove garbage
+    Removed 0 objects from db2
+
+The weakrefs still work until we remove the alias:
+
+    >>> with open('config', 'r') as f:
+    ...     db = ZODB.config.databaseFromFile(f)
+    >>> conn1 = db.open()
+    >>> conn2 = conn1.get_connection('db2')
+    >>> conn1.root.wx() is not None
+    True
+    >>> conn2.root.wx() is not None
+    True
+    >>> del conn1.root.x_alias
+    >>> transaction.commit()
+    >>> db.pack() #for d in db.databases.values(): d.pack()
+    >>> len(conn1._storage), len(conn2._storage)
+    (2, 1)
+    >>> _ = [d.close() for d in db.databases.values()]
+
+Now a GC will collect things from the database that the original
+object was in:
+
+    >>> now += 7 * 86400        # 7 days
+    >>> zc.zodbdgc.DEBUG = True
+    >>> zc.zodbdgc.gc('config', days=2, return_bad=True)
+    db1: roots
+    db1: recent
+    db2: roots
+    db2: recent
+    db1: remove garbage
+    Removed 1 objects from db1
+    db2: remove garbage
+    Removed 0 objects from db2
+    [('db1', 1)]
```

Comment:
> The fix is straightforward. Just iterate over all databases to insert the roots, ther refs and new objects, and then iterate over the databases to handle old records.

Can it be simpler? Just check if the name,oid is in `good` before removing it? Like so:
```
        for oid, tid in bad.iterator(name):
            if good.has(name, oid): # ADD THIS
                continue
            try:
                storage.deleteObject(oid, tid, t)
            except (ZODB.POSException.POSKeyError,
                    ZODB.POSException.ConflictError):
                continue
```

If I add a test case for this scenario, that change makes that test pass (and breaks my original test :)